src/main/java/com/exasol/projectkeeper/AbstractPluginPomTemplate.java

```diff
@@ -0,0 +1,168 @@
+package com.exasol.projectkeeper;
+
+import static com.exasol.projectkeeper.XPathErrorHanlingWrapper.runXpath;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.xml.XMLConstants;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.xml.sax.SAXException;
+
+/**
+ * Abstract basis for maven plugin configuration validation.
+ * <p>
+ * Create a new plugin validation by adding a template file to src/main/resources/maven_templates/. Next create an
+ * instance of this class and pass the template's name to the super constructor. If you want to enforce more that only
+ * the existence of the plugin definition, ovverride {@link #validatePluginConfiguration(Node, RunMode)}. Finally add
+ * your class to {@link PomFileTemplateRunner#TEMPLATES}.
+ * </p>
+ */
+public abstract class AbstractPluginPomTemplate implements PomTemplate {
```

Comment:
Let's refactor this class. It tries to do many things -> divergent change. 
Extract class to VerifyPom and FixPom versions. Then call these in the verify and fix methods of the runner.

Reply:
I designed it like this since it does two very coupled things: validate the structure and fix it. Both are always bound to each other since they cover the same things. Splitting this up makes it very hard to avoid code duplication and Also makes the implementation of a specific plugin mor complex. Now there a user only has to define what should be validated and the abstract basis then either fixes or validates it.

Reply:
Not sure how to split it up otherwise...

Reply:
I see the pattern of if else for runMode verify and fix. These can be in to classes with their own methods. And superclass could contain the common code.
But, we can check if makes sense again later.

Reply:
:+1: